File: Python/Memory.py
from Hardware import Hardware

class Memory(Hardware):
    __frequency = ""
    __memorySize = ""
    __supportsCuda = ""

    def __init__(self, frequency = "", memorySize = "", supportsCuda = ""):
        self.__frequency = frequency
        self.__memorySize = memorySize
        self.__supportsCuda = supportsCuda

    def setFrequency(self, frequency):
        self.__frequency = frequency
    
    def getFrequency(self):
        return self.__frequency

    def setMemorySize(self, memorySize):
        self.__memorySize = memorySize

    def getMemorySize(self):
        return self.__memorySize

    def setSupportsCuda(self, supportsCuda):
        self.__supportsCuda = supportsCuda

    def getSupportsCuda(self):
        return self.__supportsCuda

File: C++/Memory.cpp
#include <string>
#include <iostream>
#include "Hardware.cpp"

using namespace std;

class Memory : public Hardware
{
    private:
        string frequency;
        string memorySize;
        string supportsCuda;
    public:
        Memory()
        {

        }

        Memory(string frequency, string memorySize, string supportsCuda)
        {
            this->frequency = frequency;
            this->memorySize = memorySize;
            this->supportsCuda = supportsCuda;
        }

        void setFrequency(string frequency)
        {
            this->frequency = frequency;
        }

        string getFrequency()
        {
            return this->frequency;
        }

        void setMemorySize(string memorySize)
        {
            this->memorySize = memorySize;
        }

        string getMemorySize()
        {
            return this->memorySize;
        }

        void setSupportsCuda(string supportsCuda)
        {
            this->supportsCuda = supportsCuda;
        }

        string getSupportsCuda()
        {
            return this->supportsCuda;
        }

        ~Memory()
        {

        }
};